### ISSUE TITLE
[Bug]: The L0 compaction plan execution failed or trigger failed

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#30213]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-io-master-5ee9f73-20240122
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. create milvus with L0 segment
```
  config:
    dataCoord:
      enableActiveStandby: true
      segment:
        enableLevelZero: true
    indexCoord:
      enableActiveStandby: true
    log:
      level: debug
    queryCoord:
      enableActiveStandby: true
    rootCoord:
      enableActiveStandby: true
```
2. create collection with 2 shards -> index -> insert 5m-128d data -> flush -> index -> load
3. concurrent upsert + search + flush
```
'concurrent_params': {'concurrent_number': 50,
                                   'during_time': '10h',
                                   'interval': 120,
                                   'spawn_rate': None},
 'concurrent_tasks': [{'type': 'search',
                       'weight': 4,
                       'params': {'nq': 100,
                                  'top_k': 100,
                                  'search_param': {'ef': 128},
                                  'timeout': 120}},
                      {'type': 'flush',
                       'weight': 3,
                       'params': {'timeout': 120}},
                      {'type': 'upsert',
                       'weight': 4,
                       'params': {'nb': 100,
                                  'start_id': 0,
                                  'random_id': True,
                                  'random_vector': True,
                                  'timeout': 120}}]},
```
4. dataCoord segment num
![image](https://github.com/milvus-io/milvus/assets/27288593/ba755be1-9ae6-4126-afb5-e6769f0189f5)

5. From the metrics we know, L0Delete compaction executing is 0 and pending task num is increasing. 
![image](https://github.com/milvus-io/milvus/assets/27288593/9f1db15c-ce29-4e8f-95d6-827c564fca8c)

6. queryNode loaded segment num
![image](https://github.com/milvus-io/milvus/assets/27288593/2b150a76-e83f-4ff4-9824-0afcdd7cc95b)

### Milvus Log

- argo: [level-zero-stable-11a-1912554813](https://argo-workflows.zilliz.cc/archived-workflows/qa/84a91535-87df-4b5e-80b7-2ffcafa28441?nodeId=level-zero-stable-11a-1912554813)

- grafana: [metrics of level-zero-upsert-11](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-upsert-11&var-collection=All&var-app_name=milvus&from=1705915200000&to=1705965600000)

- pods:
```
level-zero-upsert-11-etcd-0                                       1/1     Running                  0               20h     10.104.15.76    4am-node20   <none>           <none>
level-zero-upsert-11-etcd-1                                       1/1     Running                  0               20h     10.104.31.47    4am-node34   <none>           <none>
level-zero-upsert-11-etcd-2                                       1/1     Running                  0               20h     10.104.28.145   4am-node33   <none>           <none>
level-zero-upsert-11-milvus-datanode-7d6b85b5d8-gxnrf             1/1     Running                  0               20h     10.104.9.6      4am-node14   <none>           <none>
level-zero-upsert-11-milvus-datanode-7d6b85b5d8-wrdv2             1/1     Running                  0               20h     10.104.14.177   4am-node18   <none>           <none>
level-zero-upsert-11-milvus-indexnode-7c7ccd8b8f-2wl2c            1/1     Running                  0               20h     10.104.33.231   4am-node36   <none>           <none>
level-zero-upsert-11-milvus-indexnode-7c7ccd8b8f-xgtmr            1/1     Running                  0               20h     10.104.12.139   4am-node17   <none>           <none>
level-zero-upsert-11-milvus-mixcoord-6b47d57ff5-d4p5r             1/1     Running                  0               20h     10.104.33.230   4am-node36   <none>           <none>
level-zero-upsert-11-milvus-proxy-5944cc9599-h9bnh                1/1     Running                  0               20h     10.104.14.178   4am-node18   <none>           <none>
level-zero-upsert-11-milvus-querynode-65d694cfdb-87h8x            1/1     Running                  0               20h     10.104.12.140   4am-node17   <none>           <none>
level-zero-upsert-11-milvus-querynode-65d694cfdb-9xbz4            1/1     Running                  0               20h     10.104.9.7      4am-node14   <none>           <none>
level-zero-upsert-11-milvus-querynode-65d694cfdb-z6s75            1/1     Running                  0               20h     10.104.16.145   4am-node21   <none>           <none>
level-zero-upsert-11-minio-0                                      1/1     Running                  0               20h     10.104.31.45    4am-node34   <none>           <none>
level-zero-upsert-11-minio-1                                      1/1     Running                  0               20h     10.104.15.78    4am-node20   <none>           <none>
level-zero-upsert-11-minio-2                                      1/1     Running                  0               20h     10.104.25.239   4am-node30   <none>           <none>
level-zero-upsert-11-minio-3                                      1/1     Running                  0               20h     10.104.32.171   4am-node39   <none>           <none>
level-zero-upsert-11-pulsar-bookie-0                              1/1     Running                  0               20h     10.104.15.82    4am-node20   <none>           <none>
level-zero-upsert-11-pulsar-bookie-1                              1/1     Running                  0               20h     10.104.31.51    4am-node34   <none>           <none>
level-zero-upsert-11-pulsar-bookie-2                              1/1     Running                  0               20h     10.104.17.111   4am-node23   <none>           <none>
level-zero-upsert-11-pulsar-bookie-init-qjtc6                     0/1     Completed                0               20h     10.104.14.168   4am-node18   <none>           <none>
level-zero-upsert-11-pulsar-broker-0                              1/1     Running                  0               20h     10.104.1.23     4am-node10   <none>           <none>
level-zero-upsert-11-pulsar-proxy-0                               1/1     Running                  0               20h     10.104.12.134   4am-node17   <none>           <none>
level-zero-upsert-11-pulsar-pulsar-init-rnspq                     0/1     Completed                0               20h     10.104.9.247    4am-node14   <none>           <none>
level-zero-upsert-11-pulsar-recovery-0                            1/1     Running                  0               20h     10.104.9.248    4am-node14   <none>           <none>
level-zero-upsert-11-pulsar-zookeeper-0                           1/1     Running                  0               20h     10.104.31.50    4am-node34   <none>           <none>
level-zero-upsert-11-pulsar-zookeeper-1                           1/1     Running                  0               20h     10.104.30.214   4am-node38   <none>           <none>
level-zero-upsert-11-pulsar-zookeeper-2                           1/1     Running                  0               20h     10.104.33.221   4am-node36   <none>           <none>
```

**Comment by XuanYang-cn** [comment]
![image](https://github.com/milvus-io/milvus/assets/51370125/8c394ea1-cd1b-4f11-9a74-eea03eec234e)
![image](https://github.com/milvus-io/milvus/assets/51370125/5bd58e4f-cd7e-46ed-8abf-defb7a8dabe1)
Mix Compaction is stucked at DataNode, might be Block by sync mgr

**Comment by XuanYang-cn** [comment]
We need to remote the injection when enable L0 segment

**Comment by XuanYang-cn** [comment]
1. Compaction Failed, not inject Done
2. Refreshed segmentBinlogs invalid

2024-01-22 18:14:34	
[2024/01/22 10:14:34.007 +00:00] [WARN] [datanode/compactor.go:530] ["compact download deltalogs wrong"] [planID=447195629143804697] [segment=447195629143586377] [path="[files/delta_log/0/0/447195629143586377/447195629136586753]"] [error="download from blob storage wrong"] [errorVerbose="download from blob storage wrong\n(1) attached stack trace\n  -- stack trace:\n  | github.com/milvus-io/milvus/internal/datanode.init\n  | \t/go/src/github.com/milvus-io/milvus/internal/datanode/binlog_io.go:40\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6525\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6502\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6502\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6502\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6502\n  | runtime.doInit\n  | \t/usr/local/go/src/runtime/proc.go:6502\n  | runtime.main\n  | \t/usr/local/go/src/runtime/proc.go:233\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1598\nWraps: (2) download from blob storage wrong\nError types: (1) *withstack.withStack (2) *errutil.leafError"]

**Comment by ThreadDao** [comment]
[compaction warn logs of delta-insert-stand-2-milvus-standalone-5d5659dbd5-xtdrq](https://grafana-4am.zilliz.cc/explore?orgId=1&left=%7B%22datasource%22:%22Loki%22,%22queries%22:%5B%7B%22refId%22:%22B%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,namespace%3D%5C%22qa-milvus%5C%22,pod%3D~%5C%22delta-insert-stand-2-milvus-standalone-5d5659dbd5-xtdrq%5C%22%7D%20%7C%3D%20%5C%22compaction%5C%22%20%7C%3D%20%5C%22WARN%5C%22%22,%22hide%22:true%7D,%7B%22refId%22:%22F%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,namespace%3D%5C%22qa-milvus%5C%22,pod%3D~%5C%22delta-insert-stand-2-milvus-standalone-5d5659dbd5-xtdrq%5C%22%7D%20%7C%3D%20%5C%22447247814132913774%5C%22%22,%22hide%22:false%7D%5D,%22range%22:%7B%22from%22:%221706115602655%22,%22to%22:%221706126481555%22%7D%7D)
- image: master-20240124-08ca0a2c
- argo: [delta-stable-stand-flush-2-3974070459](https://argo-workflows.zilliz.cc/workflows/qa/delta-stable-stand-flush-2?tab=workflow&nodeId=delta-stable-stand-flush-2-3974070459&nodePanelView=inputs-outputs)

**Comment by ThreadDao** [comment]
@XuanYang-cn It looks like the compaction won't get stuck. But I still see errors like yours in the logs. Could you please help confirm it.
[compaction logs](https://grafana-4am.zilliz.cc/explore?orgId=1&left=%7B%22datasource%22:%22Loki%22,%22queries%22:%5B%7B%22refId%22:%22B%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,namespace%3D%5C%22qa-milvus%5C%22,pod%3D~%5C%22delta-insert-stand-4-milvus-standalone-7f9bff7f68-dppwq%5C%22%7D%20%7C%3D%20%5C%22compaction%5C%22%20%7C%3D%20%5C%22WARN%5C%22%22,%22queryType%22:%22range%22,%22hide%22:true%7D,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,namespace%3D%5C%22qa-milvus%5C%22,pod%3D~%5C%22delta-insert-stand-4-milvus-standalone-7f9bff7f68-dppwq%5C%22%7D%20%7C%3D%20%5C%22447284123462416631%5C%22%22,%22queryType%22:%22range%22,%22hide%22:false%7D%5D,%22range%22:%7B%22from%22:%221706281933790%22,%22to%22:%221706283855975%22%7D%7D)
![image](https://github.com/milvus-io/milvus/assets/27288593/b2e189ec-a698-4c95-a5f7-8cbec6086099)

- image: `milvus-io-master-d87726e-20240126`
![image](https://github.com/milvus-io/milvus/assets/27288593/98ab72b7-9d31-411e-9539-e5619badc875)

**Comment by XuanYang-cn** [comment]
> @XuanYang-cn It looks like the compaction won't get stuck. But I still see errors like yours in the logs. Could you please help confirm it.

This is because compaction executed too slow and timeout canceled, will fix it soon

**Comment by XuanYang-cn** [comment]
related to #30191

**Comment by ThreadDao** [comment]
@XuanYang-cn reproduce on image: ``
- grafana: [metrics of delta-par-key-stand-5](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&from=1706789640000&to=1706868000000&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=delta-par-key-stand-5&var-collection=All&var-app_name=milvus)
- Loki: [compaction failed logs](https://grafana-4am.zilliz.cc/explore?panes=%7B%22Qa9%22:%7B%22datasource%22:%22vhI6Vw67k%22,%22queries%22:%5B%7B%22refId%22:%22B%22,%22expr%22:%22%7Bcluster%3D%5C%224am%5C%22,%20namespace%3D%5C%22qa-milvus%5C%22,%20pod%3D~%5C%22delta-par-key-stand-5-milvus-standalone-fb9c5fbf6-s4g8j%5C%22%7D%20%7C%3D%20%5C%22compaction%20task%20failed%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22vhI6Vw67k%22%7D,%22editorMode%22:%22code%22,%22hide%22:false%7D%5D,%22range%22:%7B%22from%22:%221706789640000%22,%22to%22:%221706860800000%22%7D%7D%7D&schemaVersion=1&orgId=1)
- argo: [delta-stable-stand-flush-5-3542664967](https://argo-workflows.zilliz.cc/archived-workflows/qa/2e613349-c22d-4cea-9823-0bb14b80c17f?nodeId=delta-stable-stand-flush-5-3542664967)
pods:
```
delta-par-key-stand-5-etcd-0                                      1/1     Running            0                14h     10.104.19.104   4am-node28   <none>           <none>
delta-par-key-stand-5-kafka-0                                     1/1     Running            0                14h     10.104.25.149   4am-node30   <none>           <none>
delta-par-key-stand-5-kafka-1                                     1/1     Running            0                14h     10.104.28.41    4am-node33   <none>           <none>
delta-par-key-stand-5-kafka-2                                     1/1     Running            1 (14h ago)      14h     10.104.16.138   4am-node21   <none>           <none>
delta-par-key-stand-5-kafka-zookeeper-0                           1/1     Running            0                14h     10.104.28.40    4am-node33   <none>           <none>
delta-par-key-stand-5-kafka-zookeeper-1                           1/1     Running            0                14h     10.104.16.139   4am-node21   <none>           <none>
delta-par-key-stand-5-kafka-zookeeper-2                           1/1     Running            0                14h     10.104.29.197   4am-node35   <none>           <none>
delta-par-key-stand-5-milvus-standalone-fb9c5fbf6-s4g8j           1/1     Running            5 (4h33m ago)    14h     10.104.12.102   4am-node17   <none>           <none>
delta-par-key-stand-5-minio-5d658dcc64-vt85g                      1/1     Running            0                14h     10.104.25.147   4am-node30   <none>           <none>
```

**Comment by ThreadDao** [comment]
Fixed 'master-20240202-ec688f5b-amd64'
